ansible/library/pulp_facts.py

```diff
@@ -50,27 +50,26 @@ def selinux_enforcing():
 
 
 def available_repositories(module):
-    # determine which of the platform and plugin are available,
-    # failing if platform is not available
-    available_repositories = []
+    """Find the source code for the Pulp platform and its plugins.
 
+    Raise an exception if the source code for the platform is absent.
+    """
+    available_repositories_ = []
     try:
         repos = os.listdir(module.params['devel_dir'])
     except FileNotFoundError:
         repos = []
-
-    for dirname in repos:
-        if os.path.isdir(os.path.join(module.params['devel_dir'], dirname)):
-            if dirname in repositories:
-                available_repositories.append(dirname)
-
-        # if the dirname is pulp, and it doesn't pass the isdir check, there's no
-        # platform repository in the expected location. Time to explode.
-        elif dirname == 'pulp':
-            msg = ('pulp repo not found in {} on remote machine, '
-                   'unable to proceed'.format(module.params['devel_dir']))
-            module.fail_json(name='pulp', msg=msg)
-    return available_repositories
+    for repo in repos:
+        if (os.path.isdir(os.path.join(module.params['devel_dir'], repo)) and
+                repo in repositories):
+            available_repositories_.append(repo)
+    if 'pulp' not in repos:
+        module.fail_json(name='pulp', msg=(
```

Comment:
I suppose this is a subjective thing, but I find the code much cleaner when it's separated like it was previously.